libavcodec/x86/vvc_itx_1d.asm

```diff
@@ -0,0 +1,1142 @@
+
+SECTION_RODATA
+
+const vvc_dct2_4_even_mat, dw 64, 64, 64, -64, 64, -64, 64, 64
+
+const vvc_dct2_4_odd_mat, dw 83, 36, 36, -83, -36, 83, -83, -36
+
+%define matvec_mul_4_permute(m11, m12, m13, m14, \
+                             m21, m22, m23, m24, \
+                             m31, m32, m33, m34, \
+                             m41, m42, m43, m44) \
+    m11, m12, m21, m22, m31, m32, m41, m42, \
+    m13, m14, m23, m24, m33, m34, m43, m44
+
+%define dct2_8_odd_mat_permute(m11, m12, m13, m14, \
+                               m21, m22, m23, m24, \
+                               m31, m32, m33, m34, \
+                               m41, m42, m43, m44) \
+    m41, m42, m43, m44, \
+    m31, m32, m33, m34, \
+    m21, m22, m23, m24, \
+    m11, m12, m13, m14
+    
+const vvc_dct2_8_odd_mat, dw matvec_mul_4_permute(dct2_8_odd_mat_permute( \
+    -18,  50, -75,  89, \
+    -50,  89, -18, -75, \
+    -75,  18,  89,  50, \
+    -89, -75, -50, -18))
+
+%define matvec_mul_8_permute(m11, m12, m13, m14, m15, m16, m17, m18, \
+                             m21, m22, m23, m24, m25, m26, m27, m28, \
+                             m31, m32, m33, m34, m35, m36, m37, m38, \
+                             m41, m42, m43, m44, m45, m46, m47, m48, \
+                             m51, m52, m53, m54, m55, m56, m57, m58, \
+                             m61, m62, m63, m64, m65, m66, m67, m68, \
+                             m71, m72, m73, m74, m75, m76, m77, m78, \
+                             m81, m82, m83, m84, m85, m86, m87, m88) \
+    m11, m12, m21, m22, m31, m32, m41, m42, \
+    m51, m52, m61, m62, m71, m72, m81, m82, \
+    m13, m14, m23, m24, m33, m34, m43, m44, \
+    m53, m54, m63, m64, m73, m74, m83, m84, \
+    m15, m16, m25, m26, m35, m36, m45, m46, \
+    m55, m56, m65, m66, m75, m76, m85, m86, \
+    m17, m18, m27, m28, m37, m38, m47, m48, \
+    m57, m58, m67, m68, m77, m78, m87, m88
+
+%define dct2_16_odd_mat_permute(m11, m12, m13, m14, m15, m16, m17, m18, \
+                                m21, m22, m23, m24, m25, m26, m27, m28, \
+                                m31, m32, m33, m34, m35, m36, m37, m38, \
+                                m41, m42, m43, m44, m45, m46, m47, m48, \
+                                m51, m52, m53, m54, m55, m56, m57, m58, \
+                                m61, m62, m63, m64, m65, m66, m67, m68, \
+                                m71, m72, m73, m74, m75, m76, m77, m78, \
+                                m81, m82, m83, m84, m85, m86, m87, m88) \
+    m11, m12, m13, m14, m15, m16, m17, m18, \
+    m21, m22, m23, m24, m25, m26, m27, m28, \
+    m31, m32, m33, m34, m35, m36, m37, m38, \
+    m41, m42, m43, m44, m45, m46, m47, m48, \
+    m81, m82, m83, m84, m85, m86, m87, m88, \
+    m71, m72, m73, m74, m75, m76, m77, m78, \
+    m61, m62, m63, m64, m65, m66, m67, m68, \
+    m51, m52, m53, m54, m55, m56, m57, m58
+
+const vvc_dct2_16_odd_mat, dw matvec_mul_8_permute(dct2_16_odd_mat_permute( \
```

Comment:
vvc_itx_1d.asm:89: warning: single-line macro `matvec_mul_8_permute' exists, but not taking 1 parameter [-w+pp-macro-params-single]

Reply:
See https://github.com/ffvvc/FFmpeg/pull/114#discussion_r1269083847